compiler/rustc_query_impl/src/plumbing.rs

```diff
@@ -788,3 +788,64 @@ macro_rules! define_queries {
         }
     }
 }
+
+pub fn print_query_stack<'tcx>(
+    qcx: QueryCtxt<'tcx>,
+    mut current_query: Option<QueryJobId>,
+    handler: &Handler,
+    num_frames: Option<usize>,
+) -> usize {
+    // Be careful relying on global state here: this code is called from
+    // a panic hook, which means that the global `Handler` may be in a weird
+    // state if it was responsible for triggering the panic.
+    let mut i = 0;
+
+    #[cfg(not(parallel_compiler))]
+    let query_map = qcx.try_collect_active_jobs();
```

Comment:
Can `try_collect_active_jobs` be removed from the `QueryContext` trait, and be made an inherent impl?

Reply:
I think we need to keep it, since `rustc_query_system` needs it to handle cycle errors:
https://github.com/rust-lang/rust/blob/master/compiler/rustc_query_system/src/query/plumbing.rs#L269